### PR TITLE
[REVIEW] Fix jitcache-multiprocess test failure, fix CUDA 11 related CI issues

### PR DESCRIPTION
**Pull request by harrism**
Recent changes to `rmm::mr::get_default_resource()` caused the CUDA context
to be created in the JITCACHE_MULTIPROCESS_TEST before the child process is
forked, which caused the child to not get a context.  This PR fixes the failure.

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

**Comment by galipremsagar** [comment]
rerun tests

File: CHANGELOG.md [hunk 1]
```diff
@@ -281,7 +281,7 @@
 - PR #5914 Link CUDA against libcudf_kafka
 - PR #5895 Do not break kafka client consumption loop on local client timeout
 - PR #5915 Fix reference count on Java DeviceMemoryBuffer after contiguousSplit
-
+- PR #5942 Fix JIT cache multiprocess test failure
```
Comment (davidwendt) [inline CHANGELOG.md:284]: In the rest of this file, there are 2 blank lines between the Bug section and the previous release header.

Comment (kkraus14) [inline CHANGELOG.md:284]:
```suggestion
- PR #5942 Fix JIT cache multiprocess test failure

```

**Pull request merged**